service/collector/conditions.go

```diff
@@ -0,0 +1,107 @@
+package collector
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/v2/pkg/label"
+	aeconditions "github.com/giantswarm/apiextensions/v3/pkg/conditions"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/util/conditions"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ConditionsConfig struct {
+	CtrlClient ctrlclient.Client
+	Logger     micrologger.Logger
+}
+
+type Conditions struct {
+	ctrlClient ctrlclient.Client
+	logger     micrologger.Logger
+}
+
+var (
+	clusterStatus = prometheus.NewDesc(
+		prometheus.BuildFQName(MetricsNamespace, "cluster", "status"),
+		"Latest cluster status conditions as provided by the Cluster CR status.",
+		[]string{
+			"cluster_id",
+			"release_version",
+			"status",
+		},
+		nil,
+	)
+)
+
+func NewConditions(config ConditionsConfig) (*Conditions, error) {
+	if config.CtrlClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CtrlClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	u := &Conditions{
+		ctrlClient: config.CtrlClient,
+		logger:     config.Logger,
+	}
+
+	return u, nil
+}
+
+func (u *Conditions) Collect(ch chan<- prometheus.Metric) error {
+	ctx := context.Background()
+	clusters := &capiv1alpha3.ClusterList{}
+	{
+		err := u.ctrlClient.List(ctx, clusters, ctrlclient.InNamespace(metav1.NamespaceAll))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	for _, cluster := range clusters.Items {
+		releaseVersion, ok := cluster.Labels[label.ReleaseVersion]
+		if !ok {
+			u.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("Cluster %#q has no %#q label. Skipping", cluster.Name, label.ReleaseVersion))
+			continue
+		}
+
+		var isCreating float64
+		if conditions.IsTrue(&cluster, aeconditions.CreatingCondition) {
+			isCreating = 1
+		}
+		ch <- prometheus.MustNewConstMetric(
```

Comment:
I only have one concern. Did we think about the stikiness of these metrics?
Let me explain.
It happened in the past that prometheus was remembering old metrics, even if those weren't exported since a long time.
It happened also the opposite: prometheus was "deleting" metrics because last scrape wasn't exporting them.
Which of the two is true for this metrics we are adding? Did we choose the right ones?

Reply:
This is a collector, so it's the second kind. Metrics won't stick around if the collector is not actually exposing them, which makes sense because clusters come and go.